LFS171x/docs/introduction-to-hyperledger-composer.md

```diff
@@ -0,0 +1,976 @@
+```
+concept Address {
+    o String addressLine
+    o String locality
+    o String postCode regex=/\d{4}[ ]?[A-Z]{2}/
+}
+```
+
+Finally, we define the `Fisher` and `RestaurantOwner`, which extend the Individual, and the `Regulator`:
+
+```
+participant Fisher extends Individual {
+    o String licenseNumber
+}
+
+participant RestaurantOwner extends Individual {
+    o String restaurantName
+}
+
+participant Regulator identified by id {
+    o String id
+    o String name
+}
+```
+
+## 3) Defining Assets and Transactions
+Assets and Transactions are also defined under the `models/org.tuna.cto` file.
+In `tuna-network` the asset is represented by the `Tuna`, which is defined as follow:
+
+```
+asset Tuna identified by tunaId {
+    o String tunaId
+    o Integer weight range=[500, 1000000]
+    o FishStatus status default="CAUGHT"
+    o DateTime catchTime
+    --> Individual owner
+}
+```
+
+The `Tuna` asset is uniquely identified by an id.
+It also has a weight, which is limited between 500 grams and 1 million grams (a tonne).
+The largest tuna rarely exceed 800 kg.
+
+To specify the `Status` of the Tuna, that can be either `CAUGHT` or `PURCHASED`, we can define an enumerated type `Enum`, which specifies a type that can assume a limited number of values.
+
+```
+enum FishStatus {
+    o CAUGHT
+    o PURCHASED
+}
+```
+
+Then, we define the `Transaction`, to change the ownership of the Tuna from a `Fisher` to a `RestaurantOwner`.
+```
+transaction SellTuna {
+    --> Tuna tuna
+    --> RestaurantOwner restaurantOwner
+}
+```
+
+Finally, we define the `Event` to generate after the SellTuna transaction is executed:
+```
+event TunaSale {
+    o String tunaId
+    o String restaurantName
+}
+```
+
+## 4) Developing Transaction Logic
+The transaction logic is specified in the file `lib/logic.js`.
+
+We start by defining the same namespace specified in the modeling language file.
+
+```
+'use strict';
+/**
+ * Defining the namespace for the business network
+ */
+const NS = 'org.tuna';
+```
+
+The transaction logic is defined in a function that accepts the Transaction `SellTuna` as input parameter.
+```
+/**
+* Transfer tuna from one owner to another
+* @param {org.tuna.SellTuna} tx - The transferTuna transaction
+* @transaction
+*/
+async function sellTuna(tx) {
+```
+
+Next, the registries related to the Asset `Tuna` and the Participant `Restaurant Owner` are instantiated:
+```
+    // Get asset registry for Tuna
+    const tunaRegistry = await getAssetRegistry(NS + '.Tuna');
+
+    // Get participant registry for Individuals
+    const restaurantOwnerRegistry = await getParticipantRegistry(NS + '.RestaurantOwner');
+```
+
+Next, we have to verify that the `status` of the Tuna is `CAUGHT`.
+This is to make sure that a `Tuna` already sold cannot be sold again.
+```
+    // Make sure the tuna status is CAUGHT and not PURCHASED
+    if (tx.tuna.status !== 'CAUGHT') {
```

Comment:
Can we depend on the status sent in the transaction? Or should we be checking the asset registry's value for this Tuna?

Reply:
Yes, we should be able to depend on this status, as it retrieves the object in the registry implicitly.

Reply:
Good point, will implement more explicit check and get tuna from registry.

Reply:
Done